heron/api/src/java/com/twitter/heron/api/windowing/WindowManager.java

```diff
@@ -31,15 +34,20 @@
 import static com.twitter.heron.api.windowing.EvictionPolicy.Action.PROCESS;
 import static com.twitter.heron.api.windowing.EvictionPolicy.Action.STOP;
 
-
 /**
  * Tracks a window of events and fires {@link WindowLifecycleListener} callbacks
  * on expiry of events or activation of the window due to {@link TriggerPolicy}.
  *
  * @param <T> the type of event in the window.
  */
-public class WindowManager<T> implements TriggerHandler {
+public class WindowManager<T extends Serializable> implements TriggerHandler {
   private static final Logger LOG = Logger.getLogger(WindowManager.class.getName());
+  private static final String EVICTION_STATE_KEY = "es";
```

Comment:
Will users mistakenly use the same key for their own states leading to a conflict? Same for below.

Also, should we have another separate `xxxState` class wrapping up these fields, and provide `restore(..)` and `getState(...)`?

Reply:
@maosongfu the hashmap state is partitioned into USER_STATE and WINDOWING_INTERNAL_STATE and the user will only get state.get(USER_STATE). Thus even if the user uses the same key it won't matter since its part of a different namespace as you might think of it.

The getState and restore function called in this class will in turn call the same functions for different trigger policy and eviction policy classes

Reply:
We could have another class e.g. WindowState that contain these objects but what are the benefits of that?  It just seems like a wrapper class that will only be used in the WindowManager